lib/puppet/configurer.rb

```diff
@@ -391,20 +391,18 @@ def run_internal(options)
   end
   private :run_internal
 
-  def find_functional_server
+  def find_functional_server()
     Puppet.settings[:server_list].each do |server|
       host = server[0]
       port = server[1] || Puppet[:masterport]
       begin
         http = Puppet::Network::HttpPool.http_ssl_instance(host, port)
-        response = http.get('/status/v1/simple/master')
-        return [host, port] if response.is_a?(Net::HTTPOK)
-
-        Puppet.debug(_("Puppet server %{host}:%{port} is unavailable: %{code} %{reason}") %
+        response = http.get('/status/v1/simple')
+        return [host, port] if response.is_a?(Net::HTTPOK) || response.is_a?(Net::HTTPForbidden)
+        Puppet.debug(_("Puppet server %{host}:%{port} is unavailable: % {code} %{reason}") %
                      { host: host, port: port, code: response.code, reason: response.message })
-      rescue => detail
-        #TRANSLATORS 'server_list' is the name of a setting and should not be translated
-        Puppet.debug _("Unable to connect to server from server_list setting: %{detail}") % {detail: detail}
+      rescue
+        Puppet.debug(_("Puppet server %{host}:%{port} is unreachable") % { host: host, port: port })
```

Comment:
We lost the `detail.message` about why it's unreachable. That should be restored in 5.5.x